CollAction/Controllers/AccountController.cs

```diff
@@ -118,23 +118,34 @@ public IActionResult Register(string returnUrl = null)
             if (ModelState.IsValid)
             {
                 var user = new ApplicationUser { UserName = model.Email, Email = model.Email };
-                var result = await _userManager.CreateAsync(user, model.Password);
-                if (result.Succeeded)
+                var createResult = await _userManager.CreateAsync(user, model.Password);
+                if (!createResult.Succeeded)
                 {
-                    await _newsletterSubscriptionService.SetSubscriptionAsync(model.Email, model.NewsletterSubscription);
+                    AddErrors(createResult);
+                    return View(model);
+                }
 
-                    var code = await _userManager.GenerateEmailConfirmationTokenAsync(user);
-                    var callbackUrl = Url.Action("ConfirmEmail", "Account", new { userId = user.Id, code = code }, protocol: HttpContext.Request.Scheme);
-                    await _emailSender.SendEmailAsync(model.Email, _localizer["Confirm your account"],
-                        $"{_localizer["Please confirm your account by clicking this link"]}: <a href='{callbackUrl}'>{_localizer["link"]}</a>");
-                    _logger.LogInformation(3, "User created a new account with password.");
-                    return RedirectToLocal(returnUrl);
+                // Newsletter subscription
+                await _newsletterSubscriptionService.SetSubscriptionAsync(model.Email, model.NewsletterSubscription);
+
+                // Confirmation email
+                var code = await _userManager.GenerateEmailConfirmationTokenAsync(user);
+                var callbackUrl = Url.Action("ConfirmEmail", "Account", new { userId = user.Id, code = code }, protocol: HttpContext.Request.Scheme);
+                await _emailSender.SendEmailAsync(model.Email, _localizer["Confirm your account"],
+                    $"{_localizer["Please confirm your account by clicking this link"]}: <a href='{callbackUrl}'>{_localizer["link"]}</a>");
+                _logger.LogInformation(3, "User created a new account with password.");
+
+                // Log in the new user
+                var loginResult = await _signInManager.PasswordSignInAsync(model.Email, model.Password, false, lockoutOnFailure: false);
+                if (!loginResult.Succeeded)
```

Comment:
Failing this shouldn't be possible, should be logged directly to the application error log instead of the user, who can't help this.